.github/workflows/lib-musl-tests.yml

```diff
@@ -0,0 +1,145 @@
+      - name: Parse Repository Info
+        if: github.event_name == 'repository_dispatch'
+        id: parse_repo
+        run: bash .github/scripts/utils/parse-repo.sh "${{ github.event.client_payload.pr_repo }}"
+
+      # Generate a GitHub App token for authentication
+      # This is required to create and update check runs on the target repository
+      - name: Generate GitHub App Token
+        if: github.event_name == 'repository_dispatch'
+        id: generate_token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ vars.APP_ID }}  # GitHub App ID stored in repository variables
+          private-key: ${{ secrets.PRIVATE_KEY }}  # Private key stored in repository secrets
+          owner: ${{ steps.parse_repo.outputs.owner }}
+          repositories: ${{ steps.parse_repo.outputs.repo }}
+
+      # Create an initial check run to indicate tests are starting
+      # This check run will be updated later with the final results
+      - name: Create Initial Check Run
+        if: github.event_name == 'repository_dispatch'
+        uses: octokit/request-action@v2.x
+        id: create_check
+        with:
+          route: POST /repos/{owner}/{repo}/check-runs
+          owner: ${{ steps.parse_repo.outputs.owner }}
+          repo: ${{ steps.parse_repo.outputs.repo }}
+          name: "Catalog Core Tests"
+          head_sha: ${{ github.event.client_payload.pr_sha }}
+          status: 'in_progress'
+          started_at: ${{ github.event.client_payload.start_time }}
+          output: |
+            {
+              "title": "Tests are running...",
+              "summary": "The Unikraft Catalog Core tests have been triggered. [View Progress](https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }})"
+            }
+        env:
+          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
+
+      # Checkout the test branch
+      # This branch contains the test scripts and configurations
+      - name: Checkout test branch
+        uses: actions/checkout@v4
+        with:
+          ref: test
+
+      - name: Pull .github scripts from current branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+          path: temp-main  # Isolate this checkout
+          sparse-checkout: |  # Only checkout the scripts directory
+            .github/scripts
+          sparse-checkout-cone-mode: false  # Disable cone mode for sparse checkout
+
+      # Copy the just pulled scripts to the current workspace
+      - name: Copy scripts to workspace
+        run: |
+          mkdir -p .github/scripts
+          cp -r temp-main/.github/scripts/* .github/scripts/
+          rm -rf temp-main
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Enable Docker Buildkit
+        run: bash .github/scripts/setup/docker-setup.sh
+
+      - name: Install Base Dependencies
```

Comment:
I would put this step above `Setup Docker Buildx` step so we have all the dependecies installed first then other setup